sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedSource.cs

```diff
@@ -76,7 +76,7 @@ internal DataFeedDetail InstantiateDataFeedDetail(string name, DataFeedGranulari
                     new SqlSourceParameter(d.Query) { ConnectionString = d.ConnectionString }),
                 SqlServerDataFeedSource d => new SQLServerDataFeed(name, granularityType, metricColumns, ingestionStartTime,
                     new SqlSourceParameter(d.Query) { ConnectionString = d.ConnectionString }),
-                _ => throw new InvalidOperationException("Invalid DataFeedDetail type")
+                _ => new DataFeedDetail(name, granularityType, metricColumns, ingestionStartTime) { DataSourceType = DataSourceKind }
```

Comment:
This is the "public model" => "internal model" conversion. Note that there's no need to create a `UnknownDataFeedDetail` type, because `DataFeedDetail` is not abstract. We would have needed to make it abstract if it was public, but it's used internally only.